### PR TITLE
Update applauncher masthead

### LINKED ISSUE #1031: Update Application Launcher examples to reflect latest Masthead styling
Application Launcher test pages should be updated to reflect latest design styling for Masthead here: http://www.patternfly.org/pattern-library/application-framework/masthead/#design  This includes both the[ Application Launcher for Horizontal Navigation](https://rawgit.com/patternfly/patternfly/master-dist/dist/tests/application-launcher-nav-horizontal.html) and the [Application Launcher for Vertical Navigation](https://rawgit.com/patternfly/patternfly/master-dist/dist/tests/application-launcher-nav-vertical.html). 

Specific changes include:
* For horizontal navigation -- remove the down caret "v" from the Help menu and change the container for the application selector from an InfoTip style to a standard dropdown panel.
* For vertical navigation -- make all of the above changes and also update the styling of all items to display a selected state when the menu is open.

### PR DESCRIPTION
**Pull request by mcoker**
## Description
Update masthead in app launcher examples to reflect new masthead design http://www.patternfly.org/pattern-library/application-framework/masthead/#design

fixes https://github.com/patternfly/patternfly/issues/1031

## Changes

Write a list of changes the PR introduces
* remove caret from help menu
* change app launcher dropdown menu from infotip to standard dropdown
* update app launcher's hover/open states to match the help and user links

## Link to rawgit and/or image

https://rawgit.com/michael-coker/patternfly/update-applauncher-masthead-dist/dist/tests/application-launcher.html

## PR checklist (if relevant)

- [X] **Cross browser**: works in IE9
- [x] **Cross browser**: works in IE10
- [x] **Cross browser**: works in IE11
- [x] **Cross browser**: works in Edge
- [x] **Cross browser**: works in Chrome
- [x] **Cross browser**: works in Firefox
- [x] **Cross browser**: works in Safari
- [x] **Cross browser**: works in Opera
- [x] **Responsive**: works in extra small, small, medium and large view ports.
- [x] **Preview the PR**: An image or a URL for designer to preview this PR is provided.

### CODE REVIEW

**Comment by mcoker** [comment]
@mcarrano @matthewcarleton @jeff-phillips-18 @jgiardino would someone mind reviewing this PR?

**Review by mcarrano — COMMENTED** [review]
Looks good @michael-coker except that it looks like on the tall masthead versions the User menu (i.e Brain Johnson) is lower than the two icon menus to the left. Can you check that? Might be just my eyes but they don't look vertically aligned.

**Comment by mcoker** [comment]
@mcarrano it kinda looked that way to me, too. Here are a couple of screenshots with guides that show the top and bottom of the middle (help) icon. Looks like the help and user icons are aligned to the bottom of the guide in each browser, but even though they're the same font size, they're different heights (by 1px). Want me to make an adjustment?

# firefox 
firefox

# chrome
chrome

**Comment by mcarrano** [comment]
Yes, I see what's going on @michael-coker . The other thing that adds to this illusion is that the text "Brian Johnson" is bottom aligned with the icons rather than center-aligned. Before making more changes, I'd like @kybaker to take a look. Do you feel like the user icon and/or the menu should be adjusted to make these items appear in a straight line? Or do you think it's OK as is?

**Comment by mcoker** [comment]
@mcarrano I re-worked this so the text is center aligned, but the build on Travis is failing. I believe this is the error:

    compare | Chromy error: Error. See scenario – about-modal (large-device)

I pulled down master and rebased before updating the branch, but didn't change anything with the modal. @dgutride @mindreeper2420 do you know anything about that test? You can see the build output here - https://travis-ci.org/michael-coker/patternfly/builds/378927992

**Comment by mcoker** [comment]
@mcarrano the vertical alignment updates are in the latest build if you want to review - https://rawgit.com/michael-coker/patternfly/update-applauncher-masthead-dist/dist/tests/application-launcher.html

**Review by mcarrano — DISMISSED** [review]
Looks great @michael-coker . Thanks for addressing the alignment.

File: src/less/application-launcher.less [hunk 1]
```diff
@@ -105,9 +105,8 @@
 
 .navbar-utility .applauncher-pf {
     .dropdown-menu {
-      border-width: @applauncher-pf-menu-link-border-width !important;
+      //border-width: @applauncher-pf-menu-link-border-width !important;
```
Comment (jeff-phillips-18) [inline src/less/application-launcher.less]: should this be removed?

Comment (mcoker) [reply src/less/application-launcher.less]: Good catch! Removed.

**Review by jeff-phillips-18 — APPROVED** [review]

**Comment by mcoker** [comment]
@jeff-phillips-18 @dgutride can you merge this PR?

**Pull request merged**